commit 259bd123fa4f99d8c927055d01ca7fb303168f38 (refs/changes/19/3519/1)
Author: Julien Gilli <julien.gilli@joyent.com>
Date:   2018-02-28T12:42:57-08:00 (1 year, 7 months ago)
    
    VOLAPI-99 When updating a volume, the name parameter is not being checked for.

diff --git a/test/volumes-update-name.test.js b/test/volumes-update.test.js
similarity index 92%
rename from test/volumes-update-name.test.js
rename to test/volumes-update.test.js
index 68cb3fc..28e2e69 100644
--- a/test/volumes-update-name.test.js
+++ b/test/volumes-update.test.js
@@ -127,12 +127,12 @@ if (CONFIG.experimental_cloudapi_nfs_shared_volumes !== true) {
              */
             var INVALID_NAMES = ['', '-foo', '.foo', 'x'.repeat(257)];
             vasync.forEachParallel({
-                func: function createVolume(volumeName, done) {
+                func: function updateVolume(volumeName, done) {
                     CLIENT.post('/my/volumes/' + testVolume.id, {
                         name: volumeName
-                    }, function onVolCreated(volCreatErr) {
-                        t.ok(volCreatErr, 'updating volume with name ' +
-                            volumeName + ' should error, got: ' + volCreatErr);
+                    }, function onVolUpdated(volUpdateErr) {
+                        t.ok(volUpdateErr, 'updating volume with name ' +
+                            volumeName + ' should error, got: ' + volUpdateErr);
                         done();
                     });
                 },
@@ -142,6 +142,16 @@ if (CONFIG.experimental_cloudapi_nfs_shared_volumes !== true) {
             });
         });
 
+    test('updating newly-created volume with no param should succeed',
+        function (t) {
+            CLIENT.post('/my/volumes/' + testVolume.id, {},
+                function onVolUpdated(volUpdateErr) {
+                    t.ifErr(volUpdateErr,
+                        'updating volume with no param should succeed');
+                    t.end();
+                });
+        });
+
     test('listing volumes should only include new name', function (t) {
         CLIENT.get('/my/volumes',
             function onVolumesListed(volGetErr, req, res, volumes) {
